### PR TITLE
[processor / spanmetricsprocessor] dropping the condition to replace _ with key_ as __ is reserved and _ is not

### PR DESCRIPTION
**Pull request by saikrishna397**
**Description:** when sanitizeLabel feature flag is enabled, labels that start with _ are not updated. labels that start with __ are sanitized irrespective of this flag since __ is reserved in prometheus and _ is not.

followed the same approach as #7112 & #7506

**Link to tracking Issue:**

**Testing:** emit a metric that has a label starting with _ and it should not be prefixed with key when skipSanitizeLabel is enabled.

**Documentation:**

### CODE REVIEW

**Comment by mx-psi** [comment]
@codeboten @Aneurysm9, since you reviewed the other related PRs, could you have a look at this one?

File: CHANGELOG.md [hunk 1]
```diff
@@ -220,6 +220,7 @@
 - `resourcedetectionprocessor`: add the [consul](https://www.consul.io/) detector (#6382)
 - `awsemfexporter`: refactor cw_client logic into separate `cwlogs` package (#7072)
 - `prometheusexporter`: Dropping the condition to replace _ with key_ as __ label is reserved and _ is not (#7506)
+- `spanmetricsprocessor`: Dropping the condition to replace _ with key_ as __ label is reserved and _ is not (#7506)
```
Comment (mx-psi) [inline CHANGELOG.md:223]:
```suggestion
- `spanmetricsprocessor`: Dropping the condition to replace _ with key_ as __ label is reserved and _ is not (#8057)
```

File: processor/spanmetricsprocessor/config.go [hunk 2]
```diff
@@ -58,6 +59,15 @@ type Config struct {
 	DimensionsCacheSize int `mapstructure:"dimensions_cache_size"`
 
 	AggregationTemporality string `mapstructure:"aggregation_temporality"`
+
+	// skipSanitizeLabel if enabled, labels that start with _ are not sanitized
+	skipSanitizeLabel bool
+}
+
+var dropSanitizationGate = featuregate.Gate{
+	ID:          "exporter.prometheus.PermissiveLabelSanitization",
```
Comment (mx-psi) [inline processor/spanmetricsprocessor/config.go:68]: I don't think we should have a feature gate that refers to the prometheus exporter on a different component (specially on one that can be used independently of Prometheus)

**Review by mx-psi — COMMENTED** [review]
I will defer to @codeboten and @Aneurysm9 opinions, but I think adding Prometheus-specific logic to a component that is, in principle, independent of Prometheus, is an anti-pattern that should be avoided

File: processor/spanmetricsprocessor/config.go (hunk 2 above)
Comment (Aneurysm9) [reply processor/spanmetricsprocessor/config.go:68]: Yes, this should definitely be a separate gate identifier.

File: CHANGELOG.md (hunk 1 above)
Comment (saikrishna397) [reply CHANGELOG.md]: updated.

File: processor/spanmetricsprocessor/config.go (hunk 2 above)
Comment (saikrishna397) [reply processor/spanmetricsprocessor/config.go]: @mx-psi, @Aneurysm9 updated this. please review.

**Comment by saikrishna397** [comment]
@Aneurysm9 , @mx-psi could you please review and let me know if there're any changes pending.

**Review by Aneurysm9 — APPROVED** [review]

**Review by mx-psi — APPROVED** [review]

**Review by codeboten — APPROVED** [review]

**Pull request merged**